File: submit-api/src/submit_api/schemas/account_user.py
"""AccountUserSchema class.

Manages the AccountUserSchema
"""

from marshmallow import Schema, fields

from submit_api.schemas.account import AccountSchema
from submit_api.schemas.role import UserRoleSchema


class AccountUserSchema(Schema):
    """Schema for representing an account user with roles."""

    id = fields.Int()
    invitation_id = fields.Int()
    account_id = fields.Int()
    first_name = fields.Str()
    last_name = fields.Str()
    full_name = fields.Str()
    position = fields.Str()
    work_email_address = fields.Email()
    work_contact_number = fields.Str()
    user_id = fields.Int()
    role = fields.Nested(UserRoleSchema)
    status = fields.Str(required=False)
    account = fields.Nested(AccountSchema, data_key="account", dump_only=True)


class EditRoleSchema(Schema):
    """Schema for editing a users role."""

    role_name = fields.Str()
    package_ids = fields.List(fields.Int(), allow_none=True)

File: submit-web/src/models/AccountUser.ts
import { UserPackageStatus } from "@/components/UserStatusChip";
import { Account } from "./Account";
import { USER_MANAGEMENT_ROLE } from "./Role";

export type Role = {
  account_project_id: number | null;
  package_ids: number[];
  package_names: string[];
  role_id: number;
  role_name: USER_MANAGEMENT_ROLE;
  permissions: string[];
};

export type AccountUser = {
  id: number;
  account_id: number;
  first_name: string;
  last_name: string;
  full_name: string;
  position: string;
  work_email_address: string;
  work_contact_number: string;
  account: Account;
  role: Role;
};

export type AccountUserWithRole = {
  id: number | null;
  account_id: number;
  first_name?: string;
  last_name?: string;
  full_name: string;
  position?: string;
  work_email_address: string;
  work_contact_number?: string;
  account?: Account;
  status: UserPackageStatus;
  role: Role;
  invitation_id: number;
  user_id: number | null;
};
